### PR TITLE
epic: site 4.0.0 wave 1

### PR DESCRIPTION
**Pull request by ala-n**

### CODE REVIEW

**Comment by Natalie-Smirnova** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by ala-n** [comment]
recheck cla

**Comment by ala-n** [comment]
recheck cla

File: pages/views/blogs/Prelelease v4.0.0.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+---
+layout: content
+name: Pre 4.0.0 Highlights (Part 1)
+title: Pre 4.0.0 Highlights (Part 1)
+tags: [blogs, draft]
+---
+
+Привет ребят!
+Как вы могли заметить уже начинают подключатся бета релизы ESL 4й версии.
```
Comment (dshovchko) [inline pages/views/blogs/Prelelease v4.0.0.md:9]:
```suggestion
Как вы могли заметить уже начинают подключаться бета релизы ESL 4й версии.
```

File: pages/views/blogs/Prelelease v4.0.0.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+## 1. EventUtils: listeners
+Да да, с этого релиза ESL обзавелась собственным механизмом для обработки событий.
+
+### ⚡ 1.0 Low Level API
+
+Теперь помимо `EventUtils.dispatch` доступны `EventUtils.subscribe` и `EventUtils.unsubscribe`
+
+Механизм их работы следующий:
+при помощи `EventUtils.subscribe` можно подписатся на событие
```
Comment (dshovchko) [inline pages/views/blogs/Prelelease v4.0.0.md:25]:
```suggestion
при помощи `EventUtils.subscribe` можно подписаться на событие
```

File: pages/views/blogs/Prelelease v4.0.0.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+- `EventUtils.subscribe($host, 'click', handlerFn);` или
+- `EventUtils.subscribe($host, {event: 'click', select: 'button'}, handlerFn);` или
+- `EventUtils.subscribe($host, {event: 'scroll', target: window}, handlerFn);`
+  а также множество других способов.
+
+Подробное тех описание пока доступно в [исходниках](https://github.com/exadel-inc/esl/blob/main-beta/src/modules/esl-utils/dom/events/listener.ts)
+
+Одно из основных преимуществ поверх нативного addEventListener - это расширенный контроль всех подписок.
+Все ESL листенеры сохраняются и связываются с хост элементом,
+после чего вы в любой момент можете отцепить слушателя самыми разными способвми.
```
Comment (dshovchko) [inline pages/views/blogs/Prelelease v4.0.0.md:35]:
```suggestion
после чего вы в любой момент можете отцепить слушателя самыми разными способами.
```

File: pages/views/blogs/Prelelease v4.0.0.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+- `EventUtils.unsubscribe($host, 'click');` отпишет все что привязано к $host и обрабатывает 'click'
+- `EventUtils.unsubscribe($host, 'click', handlerFn);` отпишет все что привязано к $host, обрабатывает 'click' и обрабатывается handlerFn
+- Критериев может быть любое количество.
+
+Ну a теперь самое важное и приятное связанное с этой фитчей, все что я описал выше это только Low Level API.
+Поверх заделано очень много сахара ...
+
+### ⚡ 1.1 Extended ESLBaseElement
+У всех наследников ESLBaseElement теперь есть методы `$$on` и `$$off`,
+которые имеют все те же возможности но сами выставлют хостом текущий элемент.
```
Comment (dshovchko) [inline pages/views/blogs/Prelelease v4.0.0.md:48]:
```suggestion
которые имеют все те же возможности но сами выставляют хостом текущий элемент.
```

File: pages/views/blogs/Prelelease v4.0.0.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+```typescript
+this.$$on('click', this.onClick);
+this.$$off('click'); // или this.$$off(this.onClick)
+```
+
+### ⚡ 1.2 Auto-unbind
+Все наследники ESLBaseElement отписываются от ивентов заатаченных через ESL автоматически на disconnectedCallback.
```
Comment (dshovchko) [inline pages/views/blogs/Prelelease v4.0.0.md:56]:
```suggestion
Все наследники ESLBaseElement отписываются от ивентов зааттаченных через ESL автоматически на disconnectedCallback.
```

File: pages/views/blogs/Prelelease v4.0.0.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+ ```typescript
+ @listen('click')
+ onClick(e) { ... }
+ ```
+- Доп опции также доступны
+ ```typescript
+ @listen({event: 'click', target: 'body', capture: true})
+ onBodyClick(e) { ... }
+ ```
+- Вы можете упровлять подпиской вручную, при этом связать всю или часть мета информации с самим хендлером
```
Comment (dshovchko) [inline pages/views/blogs/Prelelease v4.0.0.md:73]:
```suggestion
- Вы можете управлять подпиской вручную, при этом связать всю или часть мета информации с самим хендлером
```

File: pages/views/blogs/Prelelease v4.0.0.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+    this.$$off(this.onClick); // как и в примерах выше
+
+    EventUtils.subscribe($host, this.onClick); // Поддержка на уровне низкоуровневого API
+ }
+ ```
+
+## Mixin Components
+
+До сего момента мы могли связывать кастомные элементы ESL только с кастомным тегом.
+А что если я скажу что теперь кастомный ESL элемнт может быть связан с атрибутом?
```
Comment (dshovchko) [inline pages/views/blogs/Prelelease v4.0.0.md:90]:
```suggestion
А что если я скажу что теперь кастомный ESL элемент может быть связан с атрибутом?
```

File: pages/views/blogs/Prelelease v4.0.0.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+<my-custom-el my-attr> Может и сосуществовать с другими элементами и миксинами </my-custom-el>
+```
+будут автоматически, как и нативные кастомные теги, контролировать свой жизненный цикл и связывать его
+с зареганной MyMixin имплементацией
+
+
+## 3. `esl-utils` breaking changes
+
+Наведение красоты в `esl-utils` в процессе (и c этим связано большинство breaking changes).
+Общая стратегия ESL команды - сделать не малый обьем утилит максимально tree-shakeable.
```
Comment (dshovchko) [inline pages/views/blogs/Prelelease v4.0.0.md:121]:
```suggestion
Общая стратегия ESL команды - сделать немалый объем утилит максимально tree-shakeable.
```

File: pages/views/blogs/Prelelease v4.0.0.md [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+Соответственно утилиты переписываются в функциональный вариант:   
+`PromiseUtils.fromTimeout`(deprecated) -> `promisifyTimeout`
+За редкими исключенисями связанных механизмов, в особенности core (e.g. `EventUtils.dispatch/subscribe/unsubscribe`)
+
+По итогу велика вероятность того что все что из ESL можно будет потребить
+будет без последствий для сборки вплоть до такого варианта `import {someUtil} from '@exadel/esl';`
+
+Однако последнее пожалуй пока не рекомендовано,
+поскольку стратегия команды ESL такова что 5 версия уже будет дистрибьютатся отдельными npm пакетами:
```
Comment (dshovchko) [inline pages/views/blogs/Prelelease v4.0.0.md:130]:
```suggestion
поскольку стратегия команды ESL такова что 5 версия уже будет дистрибьютаться отдельными npm пакетами:
```

File: docs/DEVELOPMENT.md [hunk 10]
```diff
@@ -106,7 +106,7 @@ ESL project consists of the following directories:
     - [📁 _layouts](../pages/views/_layouts) - 11ty pages layouts definitions
     - [📁 components](../pages/views/components) - ESL components articles
     - [📁 examples](../pages/views/examples) - examples articles
-    - [📁 utils](../pages/views/utils) - ESL utils articles
+    - [📁 core](../pages/views/core) - ESL utils articles
```
Comment (NastaLeo) [inline docs/DEVELOPMENT.md:109]:
```suggestion
    - [📁 core](../pages/views/core) - ESL core articles
```

File: pages/views/blogs/Prelelease v4.0.0.md [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+layout: content
+name: Pre 4.0.0 Highlights (Part 1)
+title: Pre 4.0.0 Highlights (Part 1)
+tags: [blogs, draft]
+---
+
+Привет ребят!
+Как вы могли заметить уже начинают подключатся бета релизы ESL 4й версии.
+Многие из фитч еще обкатываются и не все еще доступны.
+Кроме того по итогу обязательно будет еще подробные релизноуты
```
Comment (NastaLeo) [inline pages/views/blogs/Prelelease v4.0.0.md:11]:
```suggestion
Кроме того по итогу обязательно будут еще подробные релизноуты
```

File: pages/views/blogs/Prelelease v4.0.0.md [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+(именно в формате статьи с юзкейсами и подробным описанием) прямо на [esl-ui.com](https://esl-ui.com/).
+Но тем не менее хотелось бы уже сейчас кратко осветить то что уже стало доступно,
+что можно пользовать, и какие еще аспекты библиотеке сейчас можно обновить
```
Comment (NastaLeo) [inline pages/views/blogs/Prelelease v4.0.0.md:14]:
```suggestion
что можно пользовать, и какие еще аспекты библиотеки сейчас можно обновить
```

File: pages/views/blogs/Prelelease v4.0.0.md [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+(сейчас с мажерным майлстоуном, как раз даже радикальные апдейты могут вполне оказаться одобрены и появиться в стабильной версии).
```
Comment (NastaLeo) [inline pages/views/blogs/Prelelease v4.0.0.md:15]:
```suggestion
(сейчас с мажорным майлстоуном, как раз даже радикальные апдейты могут вполне оказаться одобрены и появиться в стабильной версии).
```

File: pages/views/blogs/Prelelease v4.0.0.md [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+с зареганной MyMixin имплементацией
+
+
+## 3. `esl-utils` breaking changes
+
+Наведение красоты в `esl-utils` в процессе (и c этим связано большинство breaking changes).
+Общая стратегия ESL команды - сделать не малый обьем утилит максимально tree-shakeable.
+Соответственно утилиты переписываются в функциональный вариант:   
+`PromiseUtils.fromTimeout`(deprecated) -> `promisifyTimeout`
+За редкими исключенисями связанных механизмов, в особенности core (e.g. `EventUtils.dispatch/subscribe/unsubscribe`)
```
Comment (NastaLeo) [inline pages/views/blogs/Prelelease v4.0.0.md:124]:
```suggestion
За редкими исключениями связанных механизмов, в особенности core (e.g. `EventUtils.dispatch/subscribe/unsubscribe`)
```

**Review by NastaLeo — APPROVED** [review]

**Comment by ala-n** [comment]
Guys, you are awesome ❤️ 
But just in case the current version of the article in Russian most likely is temporary.
I'm planning to convert it to a batch of articles for all major updates that we are going present with the 4th release

**Review by nattallius — APPROVED** [review]

**Review by yadamskaya — APPROVED** [review]

File: pages/src/collection-grid/collection-grid.less [hunk 15]
```diff
@@ -6,7 +6,10 @@
   justify-content: space-around;
   grid-gap: 15px 30px;
 
+
```
Comment (fshovchko) [inline pages/src/collection-grid/collection-grid.less:9]: Unnecessary line

**Review by fshovchko — APPROVED** [review]

**Review by abarmina — APPROVED** [review]

**Pull request merged**